sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -418,6 +419,13 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
 
   override def toString: String = treeString
 
+  def toOriginString: String =
+    if (this.origin.callSite.isDefined && !this.isInstanceOf[BoundReference]) {
```

Comment:
According to my understanding, ```BoundReference``` is created during a code generation that is not correlated with an expression parsing. If we refer to ```origin``` in ```BoundReference``` that keeps the value of ```CurrentOrigin```, the result of ```toOrigingString``` may be incorrect.